ngraph/core/include/ngraph/function.hpp

```diff
@@ -138,6 +139,24 @@ namespace ngraph
         bool evaluate(const HostTensorVector& output_tensors,
                       const HostTensorVector& input_tensors) const;
 
+        /// Return a list of function's side outputs
+        const SinkVector& get_sinks(void) const { return m_sinks; }
+        /// \brief Add new sink nodes to the list to avoid removing
+        /// \param sinks new sink nodes
+        void add_sinks(const SinkVector& sinks);
+
+        /// \brief Delete sink node from the list of sinks
+        /// \param sink Sink to delete
+        void delete_sink(const std::shared_ptr<op::Sink>& sink);
+
+        /// \brief Add new Result nodes to the list to avoid removing
+        /// \param sinks new Result nodes
+        void add_results(const ResultVector& results);
```

Comment:
Use `remove` instead of `delete` following common practices in nGraph, just look at Node methods.